core/src/main/java/org/keycloak/sdjwt/vp/SdJwtVP.java

```diff
@@ -209,21 +210,28 @@ public String present(List<String> disclosureDigests, JsonNode keyBindingClaims,
     /**
      * Verifies SD-JWT presentation.
      *
-     * @param issuerSignedJwtVerificationOpts Options to parameterize the verification. A verifier must be specified
-     *                                        for validating the Issuer-signed JWT. The caller is responsible for
-     *                                        establishing trust in that associated public keys belong to the
-     *                                        intended issuer.
+     * @param issuerVerifyingKeys             Verifying keys for validating the Issuer-signed JWT. The caller
+     *                                        is responsible for establishing trust in that the keys belong
+     *                                        to the intended issuer.
+     * @param issuerSignedJwtVerificationOpts Options to parameterize the Issuer-Signed JWT verification.
      * @param keyBindingJwtVerificationOpts   Options to parameterize the Key Binding JWT verification.
      *                                        Must, among others, specify the Verifier's policy whether
      *                                        to check Key Binding.
+     * @return the fully disclosed Issuer-signed JWT's payload
      * @throws VerificationException if verification failed
      */
-    public void verify(
+    public JsonNode verify(
+            List<SignatureVerifierContext> issuerVerifyingKeys,
             IssuerSignedJwtVerificationOpts issuerSignedJwtVerificationOpts,
             KeyBindingJwtVerificationOpts keyBindingJwtVerificationOpts
     ) throws VerificationException {
-        new SdJwtVerificationContext(sdJwtVpString, issuerSignedJWT, disclosures, keyBindingJWT.orElse(null))
-                .verifyPresentation(issuerSignedJwtVerificationOpts, keyBindingJwtVerificationOpts);
+        SdJwtVerificationContext sdJwtVerificationContext = new SdJwtVerificationContext(
+                sdJwtVpString, issuerSignedJWT, disclosures, keyBindingJWT.orElse(null)
+        );
+
+        return sdJwtVerificationContext.verifyPresentation(
```

Comment:
Same as before, why does the verification return the payload and how does the return value look in case multiple credentials are included in the presentation?

Reply:
Fixed. Please refer to my comment above for the case of multiple credentials included in the presentation.